### DISCUSSION TITLE
Create generic component using useFieldArray with generic controller

### DISCUSSION THREAD

**Discussion by arshia-shakudo** [react-hook-form/react-hook-form#10047] [Q&A]
Hello everyone, I am trying to create a generic component which is using `useFieldArray` and the controller and other properties get passed from parent components. However, I have reached an error with TS about my controllers type.

```
type FormValues = {
	[x: string]: {
		key: string,
		value: string
	}[]
}
interface Props {
    prop?: boolean;
}

export default function PairInput<TFieldValues extends FormValues = FormValues,
    TFieldArrayName extends ArrayPath<TFieldValues> = ArrayPath<TFieldValues>, TKeyName extends string = "id">({
        prop,
        control,
        name,
        rules,
        shouldUnregister
    }: Props & UseFieldArrayProps<TFieldValues, TFieldArrayName, TKeyName>) {

    const { fields, append, remove } = useFieldArray<TFieldValues>({
        control,
        name: name,
    });
     

     return (
         <Controller control={control} ... />
      )
}
```

Currently in the `Controller` component I get the following error:

`Type 'Control<TFieldValues>' is not assignable to type 'Control<FieldValues>'.`

Not sure why my `Controller` component is not accepting the generic props

**Comment by bpalko-shakudo** [comment]
I believe what he is trying to do with this component is filter the 'name' prop type by keys whose corresponding value type is `{key: string, value: string`
